File: src/content/blog/porting-htomd-beyond-python/snippets/extract.py
from urllib.request import urlopen

from htomd import convert

url = (
    "https://tailwindcss.com/blog/"
    "tailwind-is-joining-shopify"
)

with urlopen(url) as response:
    html = response.read().decode("utf-8")

markdown = convert(html)
print(markdown, end="")

File: src/content/blog/porting-htomd-beyond-python/snippets/extract.rs
// HTTP client: cargo add reqwest --features blocking
use htomd::{Options, convert};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let url = concat!(
        "https://tailwindcss.com/blog/",
        "tailwind-is-joining-shopify"
    );

    let html = reqwest::blocking::get(url)?
        .error_for_status()?
        .text()?;
    let markdown = convert(&html, Options::default());

    print!("{markdown}");
    Ok(())
}
